File: backend/app/services/voice_service.py
import uuid

class VoiceService:
    def __init__(self):
        pass

    def generate(self, org_id: str, channel_id: str, user_id: str, text: str, voice_id: str | None):
        job_id = str(uuid.uuid4())
        return {"job_id": job_id, "status": "queued"}

    def status(self, job_id: str):
        return {"job_id": job_id, "status": "succeeded", "output_url": f"https://storage.local/voice/{job_id}.mp3"}

File: frontend/src/services/voiceService.js
import { apiGet, apiPost } from "./apiClient";

export async function voiceGenerate({ task_id, channel_id, voice_id, text }) {
  if (import.meta.env.VITE_AI_DEV === "1") {
    return { job_id: "voice-dev", status: "queued" };
  }
  return await apiPost("/api/v1/voice/generate", { task_id, channel_id, voice_id, text });
}

export async function voiceStatus(job_id) {
  if (import.meta.env.VITE_AI_DEV === "1") {
    return { job_id, status: "succeeded", output_url: "https://storage.local/voice/dev.mp3" };
  }
  return await apiGet(`/api/v1/voice/status/${job_id}`);
}

